frontend/src/components/RolePermissionsCard.js

```diff
@@ -0,0 +1,389 @@
+    minWidth: 400,
+    padding: "30px 20px",
+  },
+  roleName: {
+    borderRadius: 5,
+    border: "1px solid lightgray",
+    padding: theme.spacing(1),
+    paddingLeft: theme.spacing(2),
+    marginTop: theme.spacing(1),
+    marginBottom: theme.spacing(3),
+  },
+  inputField: {
+    width: "90%",
+    padding: "3px !important",
+    marginTop: theme.spacing(1),
+    marginBottom: theme.spacing(3),
+  },
+  indentedCheckbox: {
+    marginLeft: theme.spacing(2),
+  },
+  deleteButton: {
+    width: "100%",
+    color: "white",
+    marginTop: theme.spacing(2),
+  },
+  saveButton: {
+    margin: 30,
+  },
+  warning: {
+    backgroundColor: "#D15555",
+    "&:hover": {
+      backgroundColor: "#b85454",
+    },
+  },
+}));
+
+const RolePermissionCard = ({ role, mode }) => {
+  const classes = useStyles();
+  const history = useHistory();
+  const dispatch = useDispatch();
+  const [state, setState] = useState({
+    loading: false,
+    role,
+    modal: false,
+  });
+
+  const adminSubPermissions = [
+    "user_edit",
+    "project_edit",
+    "project_create",
+    "role_management",
+    "account_activation",
+  ];
+
+  const handleModalOpen = () => {
+    setState({ ...state, modal: true });
+  };
+
+  const handleModalClose = () => {
+    setState({ ...state, modal: false });
+  };
+
+  const handleDelete = () => {
+    deleteRole(state.role._id).then((res) => {
+      if (!res.ok) {
+        // reports error if server responds with invalid status
+        dispatch(openAlert(`Error: ${res.data.message}`));
+        return;
+      }
+      history.push(`/admin/roles/`);
+    });
+  };
+
+  const handleSubmit = (fun) => {
+    // checks to make sure name field is not empty
+    if (state.role.name === null || state.role.name === "") {
+      dispatch(openAlert(`Error: name field cannot be empty.`));
+      return;
+    }
+
+    // calls the create/edit function depending on the page
+    fun(state.role).then((res) => {
+      if (!res.ok) {
+        // reports error if server responds with invalid status
+        dispatch(openAlert(`Error: ${res.data.message}`));
+        return;
+      }
+      history.push(`/admin/roles/`);
+    });
+  };
+
+  const capitalize = (str) => {
+    const words = str.toLowerCase().split(" ");
+
+    for (let i = 0; i < words.length; i++) {
+      words[i] = words[i][0].toUpperCase() + words[i].substr(1);
+    }
+
+    return words.join(" ");
+  };
+
+  const handleTextField = (event) => {
+    const text = capitalize(event.target.value);
+    setState((prevState) => ({
+      ...prevState,
+      role: {
+        ...prevState.role,
+        name: text,
+      },
+    }));
+  };
+
+  const handleChange = (event) => {
+    const { permissions } = state.role;
+    const field_name = event.target.name;
+    permissions[field_name] = event.target.checked;
+    // unchecks admin box if any indented checkboxes were unchecked
+    if (permissions[field_name] !== null) {
+      permissions["admin"] = false;
+    }
```

Comment:
I'm a bit confused by this, why are we checking for `!=== null?` I assumed that event.target.checked would always be true or false. Also how do you know that permissions[field_name] is an indented field? Don't we only want to only make `permissions["admin"] = false ` if an indented box was unchecked?